File: utils/exercises/exmath.py
from mathinput import MathInput
from jinja_env import Env

class ExMath:

    def __init__(self, **kwargs):
        self.__Ex__ = "Math"
        self.question = kwargs.get('question', '')
        self.solution = kwargs.get('solution', '')
        if 'input' in kwargs:
            self.input = kwargs['input']
            self.inputblock = kwargs.get('inputblock', '')
            self.inputblock_tpl = kwargs.get('inputblock_tpl', '')
        else:
            self.input = MathInput()
            self.inputblock = self.input.render()
            self.inputblock_tpl = kwargs.get('inputblock_tpl', '')

    def set_inputblock(self, tpl, dic):
        self.inputblock_tpl = tpl
        self.inputblock = Env.from_string(tpl).render(dic)

    def update(self, dic):
        self.inputblock = Env.from_string(tpl).render(dic)

    def eval(self):
        return self.input.eval()

File: demo/math/expr/exmath.pl
extends = /model/math/expr.pl
@ /utils/exercises/exmath.py
@ /utils/inputfields/mathinput.py

tplpage =@ /model/math/template3.html

title = Calculer la distance entre deux points du plan

before == #|python|
from exmath import ExMath
ex = ExMath()
ex.question = question
ex.input.sol = sqrt(3)
ex.solution = solution
# Créer champs de réponse
#inputblock = input.render()
ex.inputblock = ex.input.render()
==

evaluator ==
ex.input.value = answers[ex.input.id]
ex.input.eval()
ex.input.display_feedback()
score = ex.input.score
ex.inputblock = ex.input.render()
==

question == 
{{ex}}
==

solution ==
La solution est ...
==

inputblock ==

==
